### PR TITLE
fix: add missing info for posthog

### LINKED ISSUE #1794: Feature: Add postHog .identify snippet to code
### Type of feature

🍕 Feature

### Current behavior

N/A

### Suggested solution

To get better tracking and be able to create cohorts, we need to setup PostHog in a way that allows us to get more context on a user.

Right now, PostHog is logging sessions anonymously, assigning a random id to users as they navigate the platform.  We can use the `posthog.identify()` method to bring more data into posthog.

What we want is to identify users to add more context to the sessions and get:
- user login (we only want their usernames, not their emails, to avoid falling into GDPR infringements) 
- whether they are a maintainer or not
- whether they are pro / free users

### Resources
From PostHog docs: 
- [Identifying users](https://posthog.com/docs/product-analytics/identify)
- [PostHog + Next.js](https://posthog.com/docs/libraries/next-js#pages-router-1)
- [PostHog + Next.js + Supabase](https://posthog.com/tutorials/nextjs-supabase-signup-funnel): this is a tutorial for a related but different thing, but the examples they use might be helpful to sort out how the three technologies work together. Getting the signup funnel created would be very helpful though, and a great extra to have if possible

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by nickytonline**
## Description

Add missing info that @isabensusan was asking for.

The additional info won't be sent everytime, just once, e.g. here's my info if you head to https://deploy-preview-1958--oss-insights.netlify.app/start

CleanShot 2023-10-19 at 15 39 18

And then subsequent calls will be the current stuff we collect, e.g. first step of onboarding.

CleanShot 2023-10-19 at 15 38 09

To make it easier on the eyeballs, filter the logs on `Posthog`

CleanShot 2023-10-19 at 15 38 49

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Fixes #1794

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

File: lib/utils/analytics.ts [hunk 1]
```diff
@@ -19,23 +19,45 @@ function initiateAnalytics() {
  * @param {string} property - The property of the event
  * @param {string} value - The value of the event
  */
-function useAnalytics() {
-  const { user } = useSupabaseAuth();
+async function captureAnalytics({ title, property, value }: AnalyticEvent) {
+  const analyticsObject: Record<string, string> = {};
```
Comment (nickytonline) [inline lib/utils/analytics.ts:23]: I put this back to a normal function. Having this as a hook was causing it to fire more than once.

File: lib/utils/analytics.ts [hunk 2]
```diff
@@ -19,23 +19,45 @@ function initiateAnalytics() {
...
+  const {
```
Comment (nickytonline) [inline lib/utils/analytics.ts:24]: I have user info retrieval happening in `captureAnalytics` atm, but it doesn't feel right as it's not the responsibility of the `captureAnalytics` functions to get the user info. It should be passed in.

To pass it in though, I need to rework some things in the app.

**Review by nickytonline — COMMENTED** [review]
This is still in draft @brandonroberts but just leaving some thoughts as I continue to refactor this.

File: pages/start.tsx [hunk 3]
```diff
@@ -45,13 +46,20 @@ interface LoginStep1Props {
...
 const LoginStep1: React.FC<LoginStep1Props> = ({ user }) => {
-  const { captureAnalytics } = useAnalytics();
+  const { data: userInfo, isLoading } = useFetchUser(user?.user_metadata.user_name);
 
-  captureAnalytics({
-    title: "User Onboarding",
-    property: "onboardingStep1",
-    value: "visited",
-  });
+  useEffect(() => {
```
Comment (nickytonline) [inline pages/start.tsx:51]: Added the `captureAnalytics` calls in `useEffect`s so they aren't called multiple times, i.e. only capture analytics once loading is complete.

Same for the other files below.

File: layouts/filter.tsx [hunk 4]
```diff
@@ -1,23 +1,59 @@
...
+  const filterOptions = useFilterOptions();
+  const topicOptions = getInterestOptions();
+  const { filterValues } = useFilterPrefetch();
+  const { toolName } = router.query;
+  const { user } = useSupabaseAuth();
+  const { data: userInfo } = useFetchUser(user?.user_metadata.user_name);
+
+  const filterBtnRouting = (filter: string) => {
+    captureAnalytics({ title: "Filters", property: "toolsFilter", value: `${filter} applied`, userInfo });
+    return router.push(`/${filterName}/${toolName}/filter/${filter.toLocaleLowerCase()}`);
```
[the remark this answers is not in the document]
Comment (nickytonline) [reply layouts/filter.tsx]: These code changes were reverted.

File: layouts/filter.tsx [hunk 5]
```diff
@@ -1,23 +1,59 @@
...
+  };
+
+  const cancelFilterRouting = () => {
+    return router.push(`/${filterName}/${toolName}`);
```
[the remark this answers is not in the document]
Comment (nickytonline) [reply layouts/filter.tsx]: These code changes were reverted.

**Review by brandonroberts — APPROVED** [review]
🧹

File: lib/utils/analytics.ts [hunk 6]
```diff
@@ -15,27 +15,29 @@ function initiateAnalytics() {
...
+    let userProperties = {};
 
-      analyticsObject[property] = value;
+    const { company, coupon_code, is_open_sauced_member, is_onboarded, role } = userInfo;
 
-      // if a user is not logged in, Posthog will generate an anonymous ID
-      if (user) {
-        posthog.identify(user.user_metadata.sub);
-      }
+    // A pro user is anyone with a role of 50 or higher
```
Comment (brandonroberts) [inline lib/utils/analytics.ts:35]: Note: This works for now, but will need updating when we move over to `tier.run`

**Review by jpmcb — APPROVED** [review]
🙈

**Pull request merged**